src/Permissions.ts

```diff
@@ -56,56 +56,73 @@ export class Permissions {
+      if (isAdminOrOwner) {
+        return true;
+      }
+
+      if (
+        // If there is an invitationId to be revoked
+        forRevokeInvitation?.invitationId
+        // If this user has created more than one invitation
+        && forRevokeInvitation?.invitationIdsCreatedByUserId?.length > 0
+        // If invitation to revoke was created by userId
+        && forRevokeInvitation?.invitationIdsCreatedByUserId.find(
+          invitationIdCreatedByUserId => invitationIdCreatedByUserId === forRevokeInvitation?.invitationId
+        )
+      ) {
+        return true;
+      }
+
+      return false;
+    }
     switch (action) {
       case Actions.ChangeSpacePlan:
-      case Actions.DeleteSpace:
-        canI = forSpace.kind !== "Personal" && this.getRole(forSpace, forUserId) === Roles.Owner;
-        break;
-      case Actions.PatchInvitation:
-      case Actions.RevokeInvitation: {
-        if ([Roles.Owner, Roles.Admin].includes(this.getRole(forSpace, forUserId))) {
-          canI = true;
-        }
+        return role === Roles.Owner
 
-        if (
-          // If there is an invitationId to be revoked
-          forRevokeInvitation?.invitationId
-          // If this user has created more than one invitation
-          && forRevokeInvitation?.invitationIdsCreatedByUserId?.length > 0
-          // If invitation to revoke was created by userId
-          && forRevokeInvitation?.invitationIdsCreatedByUserId.find(
-            invitationIdCreatedByUserId => invitationIdCreatedByUserId === forRevokeInvitation?.invitationId
-          )
-        ) {
-          canI = true;
-        }
+      case Actions.DeleteSpace:
+        return forSpace.kind !== "Personal" && role === Roles.Owner;
 
-        break;
-      }
+      case Actions.PatchInvitation:
+        return canPatchOrRevokeInvitation();
 
-      case Actions.RenameSpace: {
-        const canUpdate = [Roles.Owner, Roles.Admin].includes(this.getRole(forSpace, forUserId));
+      case Actions.RevokeInvitation:
+        return canPatchOrRevokeInvitation();
 
-        canI = forSpace.kind !== "Personal" && canUpdate;
-        break;
-      }
+      case Actions.RenameSpace:
+        return forSpace.kind !== "Personal" && isAdminOrOwner;
 
       case Actions.CreateInvitation:
+        return isAdminOrOwner;
+
       case Actions.PatchTeam:
+        return isAdminOrOwner;
+
       case Actions.PatchSpace:
+        return isAdminOrOwner;
+
       case Actions.CreateTeam:
+        return isAdminOrOwner;
+
       case Actions.DeleteTeam:
+        return isAdminOrOwner;
+
       case Actions.AddInvitationDomain:
+        return isAdminOrOwner;
+
       case Actions.GetBillingPageToken:
+        return isAdminOrOwner;
+
       case Actions.DeleteInvitationDomain:
+        return isAdminOrOwner;
+
       default:
-        canI = [Roles.Owner, Roles.Admin].includes(this.getRole(forSpace, forUserId));
-        break;
+        return isAdminOrOwner;
```

Comment:
Maybe default to false, which would expose bugs more easy. (e.g. missing action)

Reply:
good point, done